Improve deletion and cleanup of DB data

rem1niscence opened the pull request:
## Description
Improve the deletion mechanism of keys to be deleted and adds a periodic "flusher" to force BadgerDB to delete stale/deleted data

## Related Issues
<!-- List any issues this pull request closes. -->
Closes: #<issue-number>

## Changes Made
<!-- Highlight the key changes made in the code. For example:
- Added a new function to handle X
- Refactored Y for better performance
- Fixed bug Z
-->
- Add `Eviction` process to force deletion/flattening of stale data
- Fix all keys having a badger `noDiscardBit` preventing stale keys to be deleted
- Add periodic config for eviction process

## Checklist
- [x] I have tested the changes locally and verified they work as intended.
- [ ] I have appropriately titled my branch `issue-#<issue-number>`.
- [x] I have run re-built the web-wallet and/or block explorer (if applicable).
- [x] I have run `npm run prettier` to format the web-wallet and/or block explorer (if applicable)
- [ ] I have updated documentation (if applicable).
- [x] I have included tests for the changes (if applicable).

## Additional Notes
<!-- Add any additional context, screenshots, or information that reviewers might find helpful. -->
This only deletes keys using this version onwards, keys set beforehand wont be deleted, a full chain resync or snapshot from syncing this branch is recommended

andrewnguyen22 commented on store/store.go:
```diff
@@ -194,6 +196,15 @@ func (s *Store) Commit() (root []byte, err lib.ErrorI) {
 	}
+	// Trigger eviction of LSS deleted keys
+	// check if the current version is a multiple of the cleanup block interval
+	if s.Version()%s.config.StoreConfig.CleanupBlockInterval == 0 {
```
Should we have a default set if this config is 0?

rem1niscence replied:
you're right but that should be addressed on the config.go where is being set, will perfom the change there

andrewnguyen22 commented on store/store.go:
```diff
@@ -364,6 +375,112 @@ func (s *Store) setCommitID(version uint64, root []byte) lib.ErrorI {
+		return ErrCommitDB(err)
+	}
+	// this is basically a hack to force deletion of entries marked for eviction
+	// mark evict with the correct meta bits
```
Comment cleanup? Let's not use the word 'hack' let's say 'work around'

andrewnguyen22 commented on store/store.go:
```diff
@@ -194,6 +196,15 @@ func (s *Store) Commit() (root []byte, err lib.ErrorI) {
+	// check if the current version is a multiple of the cleanup block interval
+	if s.Version()%s.config.StoreConfig.CleanupBlockInterval == 0 {
+		go func() {
+			if err := s.Evict(); err != nil {
```
Is this function triggered anywhere else besides startup?

rem1niscence replied:
no, is checked after every commit and runs every X configured blocks (default 200)

andrewnguyen22 commented:
@rem1niscence reviewed your PR - looks good but it's unclear to me at this time if you are triggering Evict regularly or not... Also, should we regularly run 'GarbageCollection' to prevent these node kills we've seen?

rem1niscence commented on store/store.go:
```diff
@@ -364,6 +375,112 @@ func (s *Store) setCommitID(version uint64, root []byte) lib.ErrorI {
+		return ErrCommitDB(err)
+	}
+	// log the results
+	total, deleted := s.keyCount(lssVersion, []byte(latestStatePrefix), true)
```
I changed it to DEBUG, I think is valuable to know the amount of data set there and right now, the numbers never go past 2k tops, we may remove it if the number is bigger but is a good way to know how big our lss is getting

rem1niscence commented:
Addresses comments, also added the GC with the default discard size as I had it previously but I haven't completely benchmarked how much it affects DB operations while it runs. That number may change in the future
